libdrgn/arch_x86_64.c

```diff
@@ -623,14 +654,14 @@ linux_kernel_pgtable_iterator_next_x86_64(struct drgn_program *prog,
 			uint64_t entry = it->table[level][it->index[level]++];
 			if (bswap)
 				entry = bswap_64(entry);
-			table = entry & ADDRESS_MASK;
-			if (!(entry & PRESENT) || (entry & PSE) || level == 0) {
+			table = entry & PTE_PFN_MASK;
+			if (!(entry & _PAGE_PRESENT) || (entry & PSE) || level == 0) {
```

Comment:
Can huge pages also have numa migration entries? `pmd_present()` in the kernel checks `_PAGE_PROTNONE`, so maybe 2MB huge pages can? If so, this check probably needs to be slightly different (and depend on the level).